src/components/AssignmentTime/ExcecutionTimeMenu.tsx

```diff
@@ -0,0 +1,326 @@
+import React, { useState, useRef, useEffect, type ChangeEventHandler } from 'react'
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  Calendar,
+  Input,
+  type CalendarTypes,
+  Button,
+  Switch
+} from '@ttab/elephant-ui'
+import { timePickTypes } from './constants'
+import { useYValue } from '@/hooks/useYValue'
+import { TimeDisplay } from '../DataItem/TimeDisplay'
+import { dateToReadableDateTime } from '@/lib/datetime'
+import { useRegistry } from '@/hooks'
+import { type AssignmentData } from './types'
+interface ExcecutionTimeItemsProps extends React.PropsWithChildren {
+  handleOnSelect: ({ excecutionStart, executionEnd }: { excecutionStart: string | undefined, executionEnd: string | undefined }) => void
+  className?: string
+  index?: number
+  startDate?: string
+}
+
+const fortmatIsoStringToLocalTime = (isoString: string): JSX.Element => {
+  const date = new Date(isoString)
+  return <TimeDisplay date={date} />
+}
+
+const DateLabel = ({ fromDate, toDate }: { fromDate?: string | undefined, toDate?: string | undefined }): JSX.Element => {
+  const from = fromDate ? fortmatIsoStringToLocalTime(fromDate) : null
+  const to = toDate ? fortmatIsoStringToLocalTime(toDate) : null
+  return (
+    <span>
+      {from} {to ? '-' : ''} {to}
+    </span>
+  )
+}
+
+const testValid = (time: string): boolean => {
+  return (/^([0-1]?[0-9]|2[0-4]):([0-5][0-9])(:[0-5][0-9])?$/.test(time))
+}
+
+export const ExcecutionTimeMenu = ({ handleOnSelect, index, startDate }: ExcecutionTimeItemsProps): JSX.Element => {
```

Comment:
This is a very long component with a lot of internal state.

Reply:
Ignored for now.